ml-agents/mlagents/trainers/trainer.py

```diff
@@ -151,19 +165,58 @@ def export_model(self):
         """
         self.policy.export_model()
 
-    def write_summary(self, global_step, lesson_num=0):
+    def write_training_metrics(self):
+        """
+        Write training metrics to a CSV  file
+        :return:
+        """
+        log_file = self.summary_path + '.csv'
+        with open(log_file, 'w') as f:
+            f.writelines('%s\n' % l for l in self.training_metrics)
+
+    def write_training_metric(self, delta_update_policy,
+                              delta_train_start):
+        """
+        Write training  metrics to be eventually be persisted in a  CSV
+        :param delta_update_policy: Time (in seconds) to update policy
+        :param delta_train_start: Time (in  seconds) since training started.
+        :return:
+        """
+        logger.debug("Time to update Policy: {:0.3f} s "
+                    "Time elapsed since training: {:0.3f} s "
+                    "Time for experience collection: {:0.3f} s "
+                    "Buffer Length: {} "
+                    "Returns : {:0.3f}"
+                    .format(delta_update_policy, delta_train_start,
+                            self.delta_last_experience_collection,
+                            self.last_buffer_length,
+                            self.last_mean_return)
+                    )
+        training_metric ='{:.3f}, {:.3f}, {:.3f}, {}, {:.3f}'.format(
```

Comment:
Might want to consider using the CSV module here: https://docs.python.org/3/library/csv.html.

The CSV should also have headers -- by adding headers we might be able to also reduce duplication between the logging above and the CSV output here.  Not 100% sure about this.

Reply:
Will do